code/modules/surgery/organs/eyes.dm

```diff
@@ -140,3 +139,184 @@
 
 /obj/item/organ/eyes/robotic/shield/emp_act(severity)
 	return
+
+#define RGB2EYECOLORSTRING(definitionvar) ("[copytext(definitionvar,2,3)][copytext(definitionvar,4,5)][copytext(definitionvar,6,7)]")
+
+/obj/item/organ/eyes/robotic/glow
+	name = "High Luminosity Eyes"
+	desc = "Special glowing eyes, used by snowflakes who want to be special."
+	origin_tech = "material=3;biotech=3;engineering=3;magnets=4"
+	var/current_color_string = "#ffffff"
+	eye_color = "000"
+	var/active = FALSE
+	actions_types = list(/datum/action/item_action/organ_action/use, /datum/action/item_action/organ_action/toggle)
+	var/olddir = 0
+	var/max_light_beam_distance = 5
+	var/light_beam_distance = 5
+	var/light_object_range = 1
+	var/light_object_power = 2
+	var/list/obj/effect/abstract/eye_lighting/eye_lighting
+	var/obj/effect/abstract/eye_lighting/on_mob
+	var/image/mob_overlay
+
+/obj/item/organ/eyes/robotic/glow/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+	eye_lighting = list()
+	mob_overlay = image('icons/mob/human_face.dmi', "eyes_glow_gs")
+
+/obj/item/organ/eyes/robotic/glow/Destroy()
+	terminate_effects()
+	..()
+
+/obj/item/organ/eyes/robotic/glow/Remove()
+	terminate_effects()
+	..()
+
+/obj/item/organ/eyes/robotic/glow/proc/terminate_effects()
+	if(owner && active)
+		deactivate()
+	clear_visuals(TRUE)
+	STOP_PROCESSING(SSfastprocess, src)
+
+/obj/item/organ/eyes/robotic/glow/process()
+	if(owner && (olddir != owner.dir) && active)
+		update_visuals()
+		olddir = owner.dir
+
+/obj/item/organ/eyes/robotic/glow/ui_action_click(owner, action)
+	if(istype(action, /datum/action/item_action/organ_action/toggle))
+		toggle_active()
+	else if(istype(action, /datum/action/item_action/organ_action/use))
+		prompt_for_controls(owner)
+
+/obj/item/organ/eyes/robotic/glow/proc/toggle_active()
+	active ? deactivate() : activate()
+
+/obj/item/organ/eyes/robotic/glow/proc/prompt_for_controls(owner)
+	var/C = input(owner, "Select Color", "Select color", "#ffffff") as null|color
+	if(!C)
+		return
+	var/range = input(owner, "Enter range (0 - [max_light_beam_distance])", "Range Select", 0) as null|num
+	if(!range)
+		return
+
+	set_distance(Clamp(range, 0, max_light_beam_distance))
+	assume_rgb(C)
+
+/obj/item/organ/eyes/robotic/glow/proc/assume_rgb(newcolor)
+	current_color_string = newcolor
+	eye_color = RGB2EYECOLORSTRING(current_color_string)
+	sync_light_effects()
+	cycle_mob_overlay()
+	if(owner)
+		owner.regenerate_icons()
+
+/obj/item/organ/eyes/robotic/glow/proc/cycle_mob_overlay()
+	remove_mob_overlay()
+	sync_mob_overlay()
+	add_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/sync_mob_overlay()
```

Comment:
this proc is pointless, just use the ONE line in this proc where you use the proc.
Don't add overhead for nothing.